File: api/internal/models/system.go
package models

import "time"

type SystemVersionResponse struct {
	InstalledVersion string  `json:"installed_version"`
	LatestVersion    *string `json:"latest_version"`
	UpdateAvailable  bool    `json:"update_available"`
}

type SystemHealthStatus string

const (
	SystemStatusOK      SystemHealthStatus = "ok"
	SystemStatusWarning SystemHealthStatus = "warning"
	SystemStatusFailed  SystemHealthStatus = "failed"
	SystemStatusUnknown SystemHealthStatus = "unknown"
)

type AdminSystemHealthResponse struct {
	OverallStatus     SystemHealthStatus   `json:"overall_status"`
	GeneratedDatetime time.Time            `json:"generated_datetime"`
	API               SystemAPIHealth      `json:"api"`
	Database          SystemDatabaseHealth `json:"database"`
	Storage           SystemStorageHealth  `json:"storage"`
	Paths             SystemPathsHealth    `json:"paths"`
	Intake            SystemIntakeHealth   `json:"intake"`
	AI                SystemAIHealth       `json:"ai"`
	Frontend          SystemSimpleHealth   `json:"frontend"`
	Docker            SystemDockerHealth   `json:"docker"`
	Alerts            []SystemHealthAlert  `json:"alerts"`
}

type SystemAPIHealth struct {
	Status        SystemHealthStatus `json:"status"`
	UptimeSeconds int64              `json:"uptime_seconds"`
	ServerTime    time.Time          `json:"server_time"`
	DataRoot      string             `json:"data_root"`
	ImageRoot     string             `json:"image_root"`
	IntakeDir     string             `json:"intake_dir"`
}

type SystemDatabaseHealth struct {
	Status             SystemHealthStatus `json:"status"`
	Reachable          bool               `json:"reachable"`
	MigrationVersion   *int64             `json:"migration_version"`
	MigrationDirty     *bool              `json:"migration_dirty"`
	DatabaseSizeBytes  *int64             `json:"database_size_bytes"`
	ContainerCount     int                `json:"container_count"`
	ItemCount          int                `json:"item_count"`
	ImageAssetCount    int                `json:"image_asset_count"`
	UploadSessionCount int                `json:"upload_session_count"`
	UploadGroupCount   int                `json:"upload_group_count"`
}

type SystemStorageHealth struct {
	Status      SystemHealthStatus `json:"status"`
	Path        string             `json:"path"`
	TotalBytes  uint64             `json:"total_bytes"`
	FreeBytes   uint64             `json:"free_bytes"`
	UsedBytes   uint64             `json:"used_bytes"`
	UsedPercent float64            `json:"used_percent"`
}

type SystemPathsHealth struct {
	Status SystemHealthStatus `json:"status"`
	Paths  []SystemPathHealth `json:"paths"`
}

type SystemPathHealth struct {
	Path        string             `json:"path"`
	Status      SystemHealthStatus `json:"status"`
	Exists      bool               `json:"exists"`
	IsDirectory bool               `json:"is_directory"`
	Readable    bool               `json:"readable"`
	Writable    bool               `json:"writable"`
	Message     string             `json:"message"`
}

type SystemIntakeHealth struct {
	Status                    SystemHealthStatus `json:"status"`
	PendingOrUploadedCount    int                `json:"pending_or_uploaded_image_count"`
	ProcessingImageCount      int                `json:"processing_image_count"`
	ProcessedImageCount       int                `json:"processed_image_count"`
	FailedImageCount          int                `json:"failed_image_count"`
	StuckProcessingImageCount int                `json:"stuck_processing_image_count"`
	OldestPendingDatetime     *time.Time         `json:"oldest_pending_datetime"`
	LatestProcessedDatetime   *time.Time         `json:"latest_processed_datetime"`
	UploadSessionStatusCounts map[string]int     `json:"upload_session_status_counts,omitempty"`
}

type SystemAIHealth struct {
	Status             SystemHealthStatus `json:"status"`
	AIAssistEnabled    bool               `json:"ai_assist_enabled"`
	ActiveProviderID   *string            `json:"active_provider_id"`
	ActiveProviderName *string            `json:"active_provider_name"`
	ActiveProviderType *string            `json:"active_provider_type"`
	ActiveModelName    *string            `json:"active_model_name"`
	VisionEnabled      *bool              `json:"vision_enabled"`
	LastTestStatus     *string            `json:"last_test_status"`
	LastTestDatetime   *time.Time         `json:"last_test_datetime"`
	LastErrorMessage   *string            `json:"last_error_message"`
}

type SystemSimpleHealth struct {
	Status      SystemHealthStatus `json:"status"`
	HostingMode string             `json:"hosting_mode,omitempty"`
	PublicURL   string             `json:"public_url,omitempty"`
	Message     string             `json:"message"`
}

type SystemDockerHealth struct {
	Status            SystemHealthStatus    `json:"status"`
	Message           string                `json:"message,omitempty"`
	GeneratedDatetime *time.Time            `json:"generated_datetime,omitempty"`
	Services          []SystemDockerService `json:"services,omitempty"`
	Alerts            []SystemHealthAlert   `json:"alerts,omitempty"`
}

type SystemDockerService struct {
	ServiceName   string             `json:"service_name"`
	ContainerName string             `json:"container_name"`
	Image         string             `json:"image"`
	State         string             `json:"state"`
	Health        string             `json:"health"`
	RestartCount  int64              `json:"restart_count"`
	StartedAt     *time.Time         `json:"started_at"`
	FinishedAt    *time.Time         `json:"finished_at"`
	Ports         []string           `json:"ports"`
	Status        SystemHealthStatus `json:"status"`
}

type SystemHealthAlert struct {
	Severity SystemHealthStatus `json:"severity"`
	Area     string             `json:"area"`
	Message  string             `json:"message"`
}

File: frontend/src/types/system.ts
export type SystemHealthStatus = 'ok' | 'warning' | 'failed' | 'unknown';

export interface SystemVersionResponse {
  installed_version: string;
  latest_version: string | null;
  update_available: boolean;
}

export interface SystemHealthAlert {
  severity: 'warning' | 'failed';
  area: string;
  message: string;
}

export interface SystemAPIHealth {
  status: SystemHealthStatus;
  uptime_seconds: number;
  server_time: string;
  data_root: string;
  image_root: string;
  intake_dir: string;
}

export interface SystemDatabaseHealth {
  status: SystemHealthStatus;
  reachable: boolean;
  migration_version: number | null;
  migration_dirty: boolean | null;
  database_size_bytes: number | null;
  container_count: number;
  item_count: number;
  image_asset_count: number;
  upload_session_count: number;
  upload_group_count: number;
}

export interface SystemStorageHealth {
  status: SystemHealthStatus;
  path: string;
  total_bytes: number;
  free_bytes: number;
  used_bytes: number;
  used_percent: number;
}

export interface SystemPathCheck {
  path: string;
  status: SystemHealthStatus;
  exists: boolean;
  is_directory: boolean;
  readable: boolean;
  writable: boolean;
  message: string;
}

export interface SystemPathsHealth {
  status: SystemHealthStatus;
  paths: SystemPathCheck[];
}

export interface SystemIntakeHealth {
  status: SystemHealthStatus;
  pending_or_uploaded_image_count: number;
  processing_image_count: number;
  processed_image_count: number;
  failed_image_count: number;
  stuck_processing_image_count: number;
  oldest_pending_datetime: string | null;
  latest_processed_datetime: string | null;
  upload_session_status_counts?: Record<string, number>;
}

export interface SystemAIHealth {
  status: SystemHealthStatus;
  ai_assist_enabled: boolean;
  active_provider_id: string | null;
  active_provider_name: string | null;
  active_provider_type: string | null;
  active_model_name: string | null;
  vision_enabled: boolean | null;
  last_test_status: string | null;
  last_test_datetime: string | null;
  last_error_message: string | null;
}

export interface SystemSimpleHealth {
  status: SystemHealthStatus;
  hosting_mode?: string;
  public_url?: string;
  message: string;
}

export interface SystemDockerService {
  service_name: string;
  container_name: string;
  image: string;
  state: string;
  health: string;
  restart_count: number;
  started_at: string | null;
  finished_at: string | null;
  ports: string[];
  status: SystemHealthStatus;
}

export interface SystemDockerHealth {
  status: SystemHealthStatus;
  message?: string;
  generated_datetime?: string | null;
  services?: SystemDockerService[];
  alerts?: SystemHealthAlert[];
}

export interface AdminSystemHealthResponse {
  overall_status: SystemHealthStatus;
  generated_datetime: string;
  api: SystemAPIHealth;
  database: SystemDatabaseHealth;
  storage: SystemStorageHealth;
  paths: SystemPathsHealth;
  intake: SystemIntakeHealth;
  ai: SystemAIHealth;
  frontend: SystemSimpleHealth;
  docker: SystemDockerHealth;
  alerts: SystemHealthAlert[];
}
